src/com/sap/piper/JenkinsController.groovy

```diff
@@ -0,0 +1,75 @@
+package com.sap.piper
+
+class JenkinsController implements Serializable {
+    def script
+    def timeout = 3600
+    String jenkinsUrl = "http://localhost:8080"
+
+    def createJob(String gitUri, String jobName) {
+        String genericJobTemplate = script.libraryResource "jobTemplate.xml"
+        String jobXml = Utils.fillTemplate(genericJobTemplate, [guid: UUID.randomUUID().toString(), gitUri: gitUri])
+        return this.createJobFromJobXml(jobXml, jobName)
+    }
+
+    def waitForJenkinsStarted() {
+        def timeout = 120
+        def timePerLoop = 5
+
+        for (int i = 0; i < timeout; i += timePerLoop) {
+            script.sleep timePerLoop
+            try {
+                if (retrieveJenkinsStatus() == 'NORMAL') {
+                    return true
+                }
+            } catch (Exception e) {
+                script.echo "Could not retrieve status for Jenkins. Retrying..."
+                script.echo e.getMessage()
+                continue
+            }
+            return false
+        }
+        script.error("Timeout: Build of waiting for Jenkins status}")
+    }
+
+    JenkinsJobController createJobFromJobXml(String jobXml, String jobName) {
+        def retry = 0
+        def jobSuccessfullyCreated = false
+
+        def createJobEndpoint = BashUtils.quoteAndEscape("${jenkinsUrl}/createItem?name=${jobName}")
```

Comment:
Sounds hairy: `BashUtils.quoteAndEscape(./.)` is designed for escaping strings so that they can be used in a shell. Here we use it for escaping/encoding an URL.